app/pages/patientdata/patientdata.js

```diff
@@ -590,18 +592,20 @@ export let PatientData = React.createClass({
 
   componentWillReceiveProps: function(nextProps) {
     var userId = this.props.currentPatientInViewId;
-    var currentPatientData = _.get(this.props, ['patientDataMap', userId], null);
-
     var nextPatientData = _.get(nextProps, ['patientDataMap', userId], null);
 
-    if (!currentPatientData && nextPatientData) {
+    // Hold processing until patient is fetched (ensuring settings are accessible), AND
+    // processing hasn't already taken place (this should be cleared already when switching patients), AND
+    // nextProps patient data exists
+    if (!nextProps.fetchingPatient && !this.state.processedPatientData && nextPatientData) {
       this.doProcessing(nextProps);
     }
   },
 
   doProcessing: function(nextProps) {
     var userId = this.props.currentPatientInViewId;
     var patientData = _.get(nextProps, ['patientDataMap', userId], null);
+    var patientSettings = _.get(nextProps, ['patient', 'settings'], DEFAULT_SETTINGS);
```

Comment:
We might need to employ the `_.defaultsDeep` here as well. The `patient.settings` can be an empty object, and then `utils.processPatientData` gets angry about `settings.units.bg` not being defined.